x-pack/filebeat/fbreceiver/receiver_test.go

```diff
@@ -55,24 +78,139 @@ func TestNewReceiver(t *testing.T) {
 					FilterMessageSnippet("add_docker_metadata").
 					FilterMessageSnippet("add_kubernetes_metadata").
 					Len() == 1
-				assert.True(t, processorsLoaded, "processors not loaded")
+				assert.True(ct, processorsLoaded, "processors not loaded")
 				// Check that add_host_metadata works, other processors are not guaranteed to add fields in all environments
-				return assert.Contains(t, logs["r1"][0].Flatten(), "host.architecture")
+				return assert.Contains(ct, logs["r1"][0].Flatten(), "host.architecture")
 			}, "failed to check processors loaded")
 		},
 	})
 }
 
+func TestFactory(t *testing.T) {
+	tmpDir := t.TempDir()
+
+	monitorSocket := genSocketPath()
+	var monitorHost string
+	if runtime.GOOS == "windows" {
+		monitorHost = "npipe:///" + filepath.Base(monitorSocket)
+	} else {
+		monitorHost = "unix://" + monitorSocket
+	}
+
+	cfg := &Config{
+		Beatconfig: map[string]interface{}{
+			"filebeat": map[string]interface{}{
+				"inputs": []map[string]interface{}{
+					{
+						"type":    "benchmark",
+						"enabled": true,
+						"message": "test",
+						"count":   10,
+					},
+				},
+			},
+			"output": map[string]interface{}{
+				"otelconsumer": map[string]interface{}{},
+			},
+			"logging": map[string]interface{}{
+				"level": "info",
+				"selectors": []string{
+					"*",
+				},
+			},
+			"path.home":    tmpDir,
+			"http.enabled": true,
+			"http.host":    monitorHost,
+		},
+	}
+
+	var zapLogs bytes.Buffer
+	core := zapcore.NewCore(
+		zapcore.NewJSONEncoder(zap.NewProductionEncoderConfig()),
+		zapcore.Lock(zapcore.AddSync(&zapLogs)),
+		zapcore.InfoLevel)
+
+	factory := NewFactory()
+
+	receiverSettings := receiver.Settings{}
+	receiverSettings.Logger = zap.New(core)
+	receiverSettings.ID = component.NewIDWithName(factory.Type(), "r1")
+
+	rc, err := factory.CreateLogs(t.Context(), receiverSettings, cfg, nil)
+	require.NotEmpty(t, rc, "receiver should not be nil")
+	require.NoError(t, err)
+
+	// Ensure http metrics endpoint is reachable on receiver creation
+	var lastError strings.Builder
+	assert.Conditionf(t, func() bool {
+		return getFromSocket(t, &lastError, monitorSocket)
+	}, "failed to connect to monitoring socket, last error was: %s", &lastError)
+}
+
+func genSocketPath() string {
+	randData := make([]byte, 16)
+	for i := range len(randData) {
+		randData[i] = uint8(rand.UintN(255)) //nolint:gosec // 0-255 fits in a uint8
+	}
+	socketName := base64.URLEncoding.EncodeToString(randData) + ".sock"
+	socketDir := os.TempDir()
+	return filepath.Join(socketDir, socketName)
+}
+
+func getFromSocket(t *testing.T, sb *strings.Builder, socketPath string) bool {
+	// skip windows for now
+	if runtime.GOOS == "windows" {
+		return true
+	}
+	client := http.Client{
+		Transport: &http.Transport{
+			DialContext: func(_ context.Context, _, _ string) (net.Conn, error) {
+				return net.Dial("unix", socketPath)
+			},
+		},
+	}
+	req, err := http.NewRequestWithContext(t.Context(), http.MethodGet, "http://unix/stats", nil)
```

Comment:
can we add checking `/inputs` as well as `/stats` ?

Reply:
Absolutely.